File: python/leetcode/566-reshape-the-matrix.py
class Solution:
    def matrixReshape(self, mat: List[List[int]], r: int, c: int) -> List[List[int]]:

        new_mat = [[0 for col in range(c)] for row in range(r)]
        ROWS, COLS = len(mat), len(mat[0])

        if ROWS == 0:
            return mat
        if ROWS*COLS != r*c:
            return mat

        i, j = 0, 0

        for row in range(ROWS):
            for col in range(COLS):
                if j >= c:
                    i += 1
                    j = 0
                new_mat[i][j] = mat[row][col]
                j += 1

        return new_mat

File: c#/leetcode/566-reshape-the-matrix.cs
public class Solution {
    public int[][] MatrixReshape(int[][] mat, int r, int c) {
        int[][] newMat = new int[r][];
        
        // create the col array for each row
        for (int n=0; n < newMat.Length; n++) {
            newMat[n] = new int[c];
        }
        
        int ROWS = mat.Length;
        int COLS = mat[0].Length;
        
        if (ROWS == 0) {
            return mat;
        }
        if (r*c != ROWS*COLS) {
            return mat;
        }
        
        int i=0, j=0;
        
        for (int row=0; row<ROWS; row++) {
            
            
            for (int col=0; col<COLS;col++) {
                if (j >= c) {
                    i++;
                    j = 0;
                }
                newMat[i][j] = mat[row][col];
                j++;
            }
        }
        return newMat;
    }
}